### ISSUE TITLE
Unable to decline call

### ISSUE THREAD

**Issue by ParticleCore** [BelledonneCommunications/linphone-android#2133]
Hi, I am able to initiate and establish a call between two devices with success, but the scenario of declining an incoming call (a call that has not yet been answered) is not working correctly.

Currently using SDK 5.3.37

The following is the scenario I have tested multiple times with no success:

1. Device A calls Device B
2. Device B receives a call and prompts the user with a notification containing decline and answer options
3. Device B presses decline
4. The call is terminated on Device B correctly, and the notification is succesfully removed
5. Device A never receives anything, not a single `onCallStateChanged` event is received, so it stays stuck in the initial calling state

However, if Device A cancels the call then Device B correctly stops the incoming call, so the issue seems to be limited to declining the call on the receiving end.

I am not including logs from Device A because there aren't any entries from the moment Device A declines the call.

Below are the logs from Device B (the one that declines the call) from the moment the call is declined:

````text
Incoming call ringing for 11 seconds
Terminate CallSession [0xb400006de67d5448] which is currently in state [LinphoneCallIncomingReceived]
There is no object pool created in thread [467705179312]. Use belle_sip_object_pool_push() to create one. Unowned objects not unref'd will be leaked.
channel [0xb40000701f301e00]: message sent to [UDP://domain:5060], size: [714] bytes
SIP/2.0 603 Decline
Via: SIP/2.0/UDP domain:5060;branch=z9hG4bK-524287-1---92b5ca24e54d6244;rport
From: "testing" <sip:test2@domain>;tag=0_3119164096-765528929
To: <sip:test@domain;transport=UDP;pn-prid=<truncatred>;pn-provider=fcm;pn-param=539156446598;pn-silent=1;pn-timeout=0;pn-msg-str=IM_MSG;pn-call-str=IC_MSG;pn-groupchat-str=GC_MSG;pn-call-snd=notes_of_the_optimistic.caf;pn-msg-snd=msg.caf>;tag=K-9N~fX
Call-ID: 3119163405-765528927Y
CSeq: 1 INVITE
User-Agent: Unknown
Supported: replaces, outbound, gruu, path


Changing [server] [INVITE] transaction [0xb400006e1d6b9810], from state [PROCEEDING] to [COMPLETED]
Dialog [0xb400006e26703d90]: now updated by transaction [0xb400006e1d6b9810].
Dialog [0xb400006e26703d90] deleted (is_expired=0)
[ToneManager] notifyToneIndication reason Call declined
[ToneManager] startNamedTone
[ToneManager] Unhandled tone id.
[ToneManager] playTone playing DTMF tone 
This device (Google/Pixel 4a/sm6150) declares it has a built-in echo canceller.
No information available for [Google/Pixel 4a/sm6150],
Android >= 8 [API 33], using sound device descriptor.with DEVICE_HAS_BUILTIN_AEC | DEVICE_HAS_BUILTIN_OPENSLES_AEC flags
Sound device information for [Google/Pixel 4a/sm6150] is: builtin=[yes], delay=[0] ms
[AAudio Player] Created using device ID: AAudio Earpiece: Pixel 4a (2)
speex_lib_ctl init with neon ? 1
configuring resampler output to rate=[48000], nchannels=[1]
ms_filter_link: MSFilePlayer:0xb400006de67cc5e0,0-->MSDtmfGen:0xb400006de67cd120,0
ms_filter_link: MSDtmfGen:0xb400006de67cd120,0-->MSResample:0xb400006de674cdb0,0
ms_filter_link: MSResample:0xb400006de674cdb0,0-->MSAAudioPlayer:0xb400006de67551e0,0
Initializing speex resampler in mode [voip] from 1 channels
[AAudio Player] Using DTMF mode
[AAudio Player] Using device ID: AAudio Earpiece: Pixel 4a (2)
[AAudio Player] Player stream configured with samplerate 48000 and 1 channels
AAudioStreamBuilder_openStream() called ----------------------------------------
rate   =  48000, channels  = 1, channelMask = 0x80000001, format   = 1, sharing = EX, dir = OUTPUT
device =      2, sessionId = -1, perfMode = 12, callback: ON with frames = 0
usage  =      3, contentType = 4, inputPreset = 0, allowedCapturePolicy = 0
privacy sensitive = false, opPackageName = (null), attributionTag = (null)
Priority used: 99
Ring MSTicker priority increased to maximum.
PlayerBase::PlayerBase()
open() - openStream() returned -889, try switching from MONO to STEREO
configureDataInformation() original HW burst = 48, minMicros = 2000 => SW burst = 96
setState(s#1) from 0 to 2
configure() source format = 0x00000001, channels = 1, sink format = 0x00000001, channels = 2, useMonoBlend = 0, audioBalance = 0.000000, isExclusive 1
AAudioStreamBuilder_openStream() returns 0 = AAUDIO_OK for s#1 ----------------
[AAudio Player] Player stream opened
[AAudio Player] Expected content type 4, got 4
[AAudio Player] Expected usage 3, got 3
AAudioStream_requestStart(s#1) called --------------
setState(s#1) from 2 to 3
AAudioStream_requestStart(s#1) returned 0 ---------
[AAudio Player] Player stream started
[AAudio Player] Current state is 3 / AAUDIO_STREAM_STATE_STARTING
callbackLoop() entering >>>>>>>>>>>>>>>
onEventFromServer - got AAUDIO_SERVICE_EVENT_STARTED
setState(s#1) from 3 to 4
[AAudio Player] Waited for state change, current state is 4 / AAUDIO_STREAM_STATE_STARTED (waited for 10 ms)
CallSession [0xb400006de67d5448] moving from state LinphoneCallIncomingReceived to LinphoneCallEnd
StreamsGroup::finish() called.
=================================================================================
                         FILTER USAGE STATISTICS                                 
Name                          Count           Time/tick (ms)            CPU Usage
                                        min     mean    max     sd               
---------------------------------------------------------------------------------
MSResample                    0         0.00    0.00    0.00    0.00          0.0
MSAAudioPlayer                0         0.00    0.00    0.00    0.00          0.0
MSDtmfGen                     0         0.00    0.00    0.00    0.00          0.0
MSFilePlayer                  0         0.00    0.00    0.00    0.00          0.0
MSWebRTCAECM                  0         0.00    0.00    0.00    0.00          0.0
MSRtpSend                     0         0.00    0.00    0.00    0.00          0.0
=================================================================================
Stopping ZRTP context on session [0x0]
ZRTP context destroyed
MainDb::addEvent() of type ConferenceCallEnded (value 4)
Update conference call in database: 3119163405-765528927Y
Linphone core [0xb400006e866ef920] notified [call_log_updated]
Resetting the current call
Removing the call (local address sip:test@domain remote address "testing" <sip:test2@domain>) from the list attached to the core
Notifying capture sound card that is no longer needed.
Notifying playback sound card that is no longer needed.
[Core Manager] Last call ended
[Audio Helper] Ringing audio focus request abandonned
resetDrmState:  mDrmInfo=null mDrmProvisioningThread=null mPrepareDrmInProgress=false mActiveDrmScheme=false
cleanDrmObj: mDrmObj=null mDrmSessionId=null
resetDrmState:  mDrmInfo=null mDrmProvisioningThread=null mPrepareDrmInProgress=false mActiveDrmScheme=false
cleanDrmObj: mDrmObj=null mDrmSessionId=null
[Audio Helper] Ringtone ringing stopped
[Audio Helper] Call audio focus request was already abandonned
[Core Service] Last call ended
[Core Service] Stopping service as foreground
[Core Manager] CoreService is no longer running in foreground
[Core Service] Last call ended
Linphone core [0xb400006e866ef920] notified [last_call_ended]
Unable to find audio video conference with conference ID ConferenceId(peer=sip:test@domain;transport=UDP;pn-prid=<truncated>;pn-provider=fcm;pn-param=539156446598;pn-silent=1;pn-timeout=0;pn-msg-str=IM_MSG;pn-call-str=IC_MSG;pn-groupchat-str=GC_MSG;pn-call-snd=notes_of_the_optimistic.caf;pn-msg-snd=msg.caf, local=sip:test@domain;transport=UDP;pn-prid=<truncated>;pn-provider=fcm;pn-param=539156446598;pn-silent=1;pn-timeout=0;pn-msg-str=IM_MSG;pn-call-str=IC_MSG;pn-groupchat-str=GC_MSG;pn-call-snd=notes_of_the_optimistic.caf;pn-msg-snd=msg.caf) in RAM.
onCallStateChanged End Call terminated
[Core Service] Stopping vibrator
Linphone core [0xb400006e866ef920] notified [call_state_changed]
bellesip_wake_lock_acquire(): Android wake lock [belle-sip recv channel] acquired [ref=0x478e]
bellesip_wake_lock_acquire(): cast long of wakelock 18318
channel [0xb40000701f301e00]: starting recv background task with id=[478e].
channel [0xb40000701f301e00]: received [1135] new bytes from [UDP://domain:5060]:
ACK sip:test@domain;transport=UDP;pn-prid=<truncated>;pn-provider=fcm;pn-param=539156446598;pn-silent=1;pn-timeout=0;pn-msg-str=IM_MSG;pn-call-str=IC_MSG;pn-groupchat-str=GC_MSG;pn-call-snd=notes_of_the_optimistic.caf;pn-msg-snd=msg.caf SIP/2.0
Via: SIP/2.0/UDP domain:5060;branch=z9hG4bK-524287-1---92b5ca24e54d6244;rport
Max-Forwards: 70
To: <sip:test@domain;transport=UDP;pn-prid=<truncated>;pn-provider=fcm;pn-param=539156446598;pn-silent=1;pn-timeout=0;pn-msg-str=IM_MSG;pn-call-str=IC_MSG;pn-groupchat-str=GC_MSG;pn-call-snd=notes_of_the_optimistic.caf;pn-msg-snd=msg.caf>;tag=K-9N~fX
From: "testing"<sip:test2@domain>;tag=0_3119164096-765528929
Call-ID: 3119163405-765528927Y
CSeq: 1 ACK
User-Agent:  Mitel
Content-Length: 0


channel [0xb40000701f301e00] [1135] bytes parsed
Found transaction [0xb400006e1d6b9810] matching request.
Changing [server] [INVITE] transaction [0xb400006e1d6b9810], from state [COMPLETED] to [CONFIRMED]
channel [0xb40000701f301e00]: ending recv background task with id=[478e].
bellesip_wake_lock_release(): Android wake lock released [ref=0x478e]
Dialog [0xb400006e26703d90] terminated for op [0xb400006e76ae20c0]
CallSession [0xb400006de67d5448] moving from state LinphoneCallEnd to LinphoneCallReleased
Unable to find audio video conference with conference ID ConferenceId(peer=sip:test@domain;transport=UDP;pn-prid=<truncated>;pn-provider=fcm;pn-param=539156446598;pn-silent=1;pn-timeout=0;pn-msg-str=IM_MSG;pn-call-str=IC_MSG;pn-groupchat-str=GC_MSG;pn-call-snd=notes_of_the_optimistic.caf;pn-msg-snd=msg.caf, local=sip:test@domain;transport=UDP;pn-prid=<truncated>;pn-provider=fcm;pn-param=539156446598;pn-silent=1;pn-timeout=0;pn-msg-str=IM_MSG;pn-call-str=IC_MSG;pn-groupchat-str=GC_MSG;pn-call-snd=notes_of_the_optimistic.caf;pn-msg-snd=msg.caf) in RAM.
onCallStateChanged Released Call released
Linphone core [0xb400006e866ef920] notified [call_state_changed]
op [0xb400006e76ae20c0] : setOrUpdateDialog() current=[0xb400006e26703d90] new=[0x0]
RingStream no longer needed.
[ToneManager] destroyRingStream
[AAudio Player] Closing player stream, current state is 4
AAudioStream_close(s#1) called ---------------
release_l(): mServiceStreamHandle = 0x00000021
callbackLoop() exiting, result = 96, isActive() = 1 <<<<<<<<<<<<<<
joinThread_l() pthread_join succeeded
stop(nanos = 18901337691831) max lateness = 938 micros
setState(s#1) from 4 to 9
setState(s#1) from 9 to 11
setState(s#1) from 11 to 11
setState(s#1) from 11 to 12
~AudioStreamInternal() 0xb400006ea6723740 called
~AudioStream(s#1) mPlayerBase strongCount = 2
AAudioStream_close(s#1) returned 0 ---------
[AAudio Player] Player stream closed
ms_filter_unlink: MSFilePlayer:0xb400006de67cc5e0,0-->MSDtmfGen:0xb400006de67cd120,0
ms_filter_unlink: MSDtmfGen:0xb400006de67cd120,0-->MSResample:0xb400006de674cdb0,0
ms_filter_unlink: MSResample:0xb400006de674cdb0,0-->MSAAudioPlayer:0xb400006de67551e0,0
Ring MSTicker thread exiting
````

Methods used to terminate the call:

````kotlin
core?.calls?.firstOrNull()?.terminate()
core?.currentCall?.terminate()
core?.currentCall?.decline(Reason.Declined)
core?.terminateAllCalls()
````

With each one respectively returning the following:

````kotlin
0
null
null
0
````

This issue sounds very similar to what was already reported in #2005 

Is there something being done incorrectly here or is this a bug? I could not find any documentation explaining how this is suppose to be done so I followed the tutorial.

**Comment by ParticleCore** [comment]
Just an added note: I also tried testing the exact same scenario using the Linphone Android app available on the Play Store (installed on both devices) and the result was exactly the same; declining a call on the receiving end does not terminate the call on the initiator side.

Linphone Android App: 5.2.3 with SDK 5.3.14

**Comment by Viish** [comment]
Hi,

Without the logs of the device A, it's hard to say for sure, but given you are using UDP and are using push notifications (that generate a long token sent in the Contact header) there is a very high probability your packet will never reach it's destination.
You can do the test by disabling push notifications, it's more likely to succeed.
But the only real solution here is to switch to TCP or even better TLS as transport protocol.
SIP over UDP is a very bad thing to do.

Cheers,

**Comment by pookie13** [comment]
I also agree with @Viish answer: When using TCP or TLS, you receive a callback indicating whether your packet reached its destination, whereas UDP does not guarantee packet delivery to the intended destination.

**Comment by ParticleCore** [comment]
Same thing happens with TCP (could not test with TLS yet).

Below are the logs from device A:

````text
New MediaSession [0xb400007581539eb8] initialized (liblinphone version: 5.3.37)
Rtp bundle is disabled.
[ToneManager] stopDtmfStream
Linphone core [0xb4000076214e33d0] notified [call_created]
Notifying capture sound card that it is going to be used.
Notifying playback sound card that it is going to be used.
CallSession [0xb400007581539eb8] moving from state LinphoneCallIdle to LinphoneCallOutgoingInit
[Core Manager] First call started
[Core Manager] Starting service org.linphone.core.tools.service.CoreService
[Core Service] First call started
[Core Service] Starting service as foreground
[Core Manager] CoreService is now running in foreground
[Core Service] First call started
Linphone core [0xb4000076214e33d0] notified [first_call_started]
[Core Manager] Outgoing call in progress, no other call, acquire ringing audio focus for ringback
[Audio Helper] Ringing audio focus request still active, skipping
Linphone core [0xb4000076214e33d0] notified [call_state_changed]
Found media local-ip from signaling connection: 192.168.2.19
Guessed media local ip address is 192.168.2.19
BctbxException occurred: 
[LIME]  ../../../lime/src/lime_localStorage.cpp:269 Cannot find Lime User sip:test@server:41822;transport=TCP in DB while setting up identity key for ZRTP auxiliary secret
stream#0 [audio] in state [Stopped]: multicast role is [inactive]
RtpSession bound to [::0] ports [40566] [42233]
Fail to increase socket's recv buffer size with SO_RCVBUFFORCE: Operation not permitted.
Fail to increase socket's send buffer size with SO_SNDBUFFORCE: Operation not permitted.
Fail to increase socket's recv buffer size with SO_RCVBUFFORCE: Operation not permitted.
Configured srtp crypto suite: AEAD_AES_128_GCM 
Configured srtp crypto suite: AES_CM_128_HMAC_SHA1_80 
Configured srtp crypto suite: AEAD_AES_256_GCM 
Configured srtp crypto suite: AES_256_CM_HMAC_SHA1_80 
Creating ZRTP engine on rtp session [0xb4000076815189c0] ssrc 0xf45eb9b3
rtp_session_enable_network_simulation:DISABLING NETWORK SIMULATION
Setting DSCP to 46 for MSAudio stream.
Created stream of type audio at index 0: stream#0 [audio] in state [Stopped]
makeLocalMediaDescription: address = 192.168.2.19
Local media description assigned to op 0xb4000076115dbc00
Created local media description.
AudioStream[0xb4000076014e2ab0]: mic is [enabled].
This device (Google/Pixel 4a/sm6150) declares it has a built-in echo canceller.
No information available for [Google/Pixel 4a/sm6150],
Android >= 8 [API 33], using sound device descriptor.with DEVICE_HAS_BUILTIN_AEC | DEVICE_HAS_BUILTIN_OPENSLES_AEC flags
Sound device information for [Google/Pixel 4a/sm6150] is: builtin=[yes], delay=[0] ms
Could not apply gain on sent RTP packets: gain control wasn't activated. Use audio_stream_enable_gain_control() before starting the stream.
[MS2AudioStream] setting type of soundcard 0xb4000075714ca930 to voice
Call's input audio device is Pixel 4a
[MS2AudioStream] setting type of soundcard 0xb4000075714d37d0 to voice
[ToneManager] stopDtmfStream
ICE state is IceStateNotActivated for stream#0 [audio] in state [Stopped]
Contact Address(<sip:test@server:41822;transport=TCP>;+sip.instance="<urn:uuid:<truncated>>";expires=3600;+org.linphone.specs="lime") has been fixed using account
Setting contact address for session 0xb4000076115c4d70 to Address(<sip:test@server:41822;transport=TCP>;+sip.instance="<urn:uuid:<truncated>>";expires=3600;+org.linphone.specs="lime")
[sip:test@server] calling [sip:test2@server] on op [0xb4000076115dbc00]
Skipping top route of initial route-set because same as request-uri
bellesip_wake_lock_acquire(): Android wake lock [belle-sip transaction(0xb4000075b15392f0)] acquired [ref=0x4aa6]
bellesip_wake_lock_acquire(): cast long of wakelock 19110
transaction [0xb4000075b15392f0]: starting transaction background task with id=[4aa6].
Changing [client] [INVITE] transaction [0xb4000075b15392f0], from state [INIT] to [CALLING]
channel [0xb4000077be1f0e10]: message sent to [TCP://server:5060], size: [1118] bytes
INVITE sip:test2@server SIP/2.0
Via: SIP/2.0/TCP 192.168.2.19:41822;branch=z9hG4bK.layOIyBZ3;rport
From: <sip:test@server>;tag=Ls-v5FPWU
To: sip:test2@server
CSeq: 20 INVITE
Call-ID: tSdngCajPw
Max-Forwards: 70
Supported: replaces, outbound, gruu, path
Allow: INVITE, ACK, CANCEL, OPTIONS, BYE, REFER, NOTIFY, MESSAGE, SUBSCRIBE, INFO, PRACK, UPDATE
Content-Type: application/sdp
Content-Length: 506
Contact: <sip:test@server:41822;transport=tcp>;+sip.instance="<urn:uuid:<truncated>>";expires=3600;+org.linphone.specs="lime"
User-Agent: Unknown

v=0
o=554 893 3297 IN IP4 192.168.2.19
s=Talk
c=IN IP4 192.168.2.19
t=0 0
a=rtcp-xr:rcvr-rtt=all:10000 stat-summary=loss,dup,jitt,TTL voip-metrics
m=audio 40566 RTP/AVP 96 97 98 0 8 101 99 100
a=rtpmap:96 opus/48000/2
a=fmtp:96 useinbandfec=1
a=rtpmap:97 speex/16000
a=fmtp:97 vbr=on
a=rtpmap:98 speex/8000
a=fmtp:98 vbr=on
a=rtpmap:101 telephone-event/48000
a=rtpmap:99 telephone-event/16000
a=rtpmap:100 telephone-event/8000
a=rtcp:42233
a=rtcp-fb:* trr-int 1000
a=rtcp-fb:* ccm tmmbr

CallSession [0xb400007581539eb8] moving from state LinphoneCallOutgoingInit to LinphoneCallOutgoingProgress
Unable to find audio video conference with conference ID ConferenceId(peer=sip:test2@server, local=sip:test2@server) in RAM.
Linphone core [0xb4000076214e33d0] notified [call_state_changed]
MainDb::addEvent() of type ConferenceCallStarted (value 3)
Insert new conference call in database: tSdngCajPw
Linphone core [0xb4000076214e33d0] notified [call_log_updated]
[Core Service] Started
bellesip_wake_lock_acquire(): Android wake lock [belle-sip recv channel] acquired [ref=0x4b06]
bellesip_wake_lock_acquire(): cast long of wakelock 19206
channel [0xb4000077be1f0e10]: starting recv background task with id=[4b06].
channel [0xb4000077be1f0e10]: received [864] new bytes from [TCP://server:5060]:
SIP/2.0 100 Trying
Via: SIP/2.0/TCP 192.168.2.19:41822;branch=z9hG4bK.layOIyBZ3;rport=41822;received=server
To: <sip:test2@server>
From: <sip:test@server>;tag=Ls-v5FPWU
Call-ID: tSdngCajPw
CSeq: 20 INVITE
User-Agent:  Mitel
Content-Length: 0

SIP/2.0 401 Unauthorized
Via: SIP/2.0/TCP 192.168.2.19:41822;branch=z9hG4bK.layOIyBZ3;rport=41822;received=server
Contact: <sip:test2@server:5060;transport=TCP>
To: <sip:test2@server>;tag=57544348
From: <sip:test@server>;tag=Ls-v5FPWU
Call-ID: tSdngCajPw
CSeq: 20 INVITE
Date: Mon, 01 Apr 2024 13:40:49 GMT
User-Agent:  Mitel
WWW-Authenticate: Digest nonce="1711978849:fb65ca03b7e04f3908dc43b682b2ab25",algorithm=MD5,realm="mbg",qop="auth,auth-int"
Content-Length: 0


channel [0xb4000077be1f0e10] [315] bytes parsed
channel [0xb4000077be1f0e10] [549] bytes parsed
Found transaction matching response.
Changing [client] [INVITE] transaction [0xb4000075b15392f0], from state [CALLING] to [PROCEEDING]
op [0xb4000076115dbc00] : setOrUpdateDialog() current=[0x0] new=[0x0]
Op [0xb4000076115dbc00] receiving call response [100], dialog is [0x0] in state [BELLE_SIP_DIALOG_NULL]
Found transaction matching response.
Changing [client] [INVITE] transaction [0xb4000075b15392f0], from state [PROCEEDING] to [COMPLETED]
channel [0xb4000077be1f0e10]: message sent to [TCP://server:5060], size: [426] bytes
ACK sip:test2@server SIP/2.0
Via: SIP/2.0/TCP 192.168.2.19:41822;branch=z9hG4bK.layOIyBZ3;rport
Call-ID: tSdngCajPw
From: <sip:test@server>;tag=Ls-v5FPWU
To: <sip:test2@server>;tag=57544348
Contact: <sip:test@server:41822;transport=tcp>;+sip.instance="<urn:uuid:<truncated>>";expires=3600;+org.linphone.specs="lime"
Max-Forwards: 70
CSeq: 20 ACK
Content-Length: 0


linphone_core_find_auth_info(): returning auth info username=test, realm=mbg
AuthStack::authFound() for Username[test];Userid[];Realm[mbg];Domain[server];Algorithm[];AvailableAlgorithms[];
Auth info found for [test] realm [mbg]
bellesip_wake_lock_acquire(): Android wake lock [belle-sip transaction(0xb4000075b14fe1f0)] acquired [ref=0x4b46]
bellesip_wake_lock_acquire(): cast long of wakelock 19270
transaction [0xb4000075b14fe1f0]: starting transaction background task with id=[4b46].
Changing [client] [INVITE] transaction [0xb4000075b14fe1f0], from state [INIT] to [CALLING]
channel [0xb4000077be1f0e10]: message sent to [TCP://server:5060], size: [1364] bytes
INVITE sip:test2@server SIP/2.0
Via: SIP/2.0/TCP 192.168.2.19:41822;branch=z9hG4bK.7m2dpH0Mm;rport
From: <sip:test@server>;tag=Ls-v5FPWU
To: sip:test2@server
CSeq: 21 INVITE
Call-ID: tSdngCajPw
Max-Forwards: 70
Supported: replaces, outbound, gruu, path
Allow: INVITE, ACK, CANCEL, OPTIONS, BYE, REFER, NOTIFY, MESSAGE, SUBSCRIBE, INFO, PRACK, UPDATE
Content-Type: application/sdp
Content-Length: 506
Contact: <sip:test@server:41822;transport=tcp>;+sip.instance="<urn:uuid:<truncated>>";expires=3600;+org.linphone.specs="lime"
User-Agent: Unknown
Authorization:  Digest realm="mbg", nonce="1711978849:fb65ca03b7e04f3908dc43b682b2ab25", algorithm=MD5, username="test",  uri="sip:test2@server", response="8abd5fef8893a3ac853eeee48ce4faa5", cnonce="tzQxevdBghi5C7iX", nc=00000001, qop=auth

v=0
o=554 893 3297 IN IP4 192.168.2.19
s=Talk
c=IN IP4 192.168.2.19
t=0 0
a=rtcp-xr:rcvr-rtt=all:10000 stat-summary=loss,dup,jitt,TTL voip-metrics
m=audio 40566 RTP/AVP 96 97 98 0 8 101 99 100
a=rtpmap:96 opus/48000/2
a=fmtp:96 useinbandfec=1
a=rtpmap:97 speex/16000
a=fmtp:97 vbr=on
a=rtpmap:98 speex/8000
a=fmtp:98 vbr=on
a=rtpmap:101 telephone-event/48000
a=rtpmap:99 telephone-event/16000
a=rtpmap:100 telephone-event/8000
a=rtcp:42233
a=rtcp-fb:* trr-int 1000
a=rtcp-fb:* ccm tmmbr

channel [0xb4000077be1f0e10]: ending recv background task with id=[4b06].
bellesip_wake_lock_release(): Android wake lock released [ref=0x4b06]
bellesip_wake_lock_acquire(): Android wake lock [belle-sip recv channel] acquired [ref=0x46a2]
bellesip_wake_lock_acquire(): cast long of wakelock 18082
channel [0xb4000077be1f0e10]: starting recv background task with id=[46a2].
channel [0xb4000077be1f0e10]: received [315] new bytes from [TCP://server:5060]:
SIP/2.0 100 Trying
Via: SIP/2.0/TCP 192.168.2.19:41822;branch=z9hG4bK.7m2dpH0Mm;rport=41822;received=server
To: <sip:test2@server>
From: <sip:test@server>;tag=Ls-v5FPWU
Call-ID: tSdngCajPw
CSeq: 21 INVITE
User-Agent:  Mitel
Content-Length: 0
channel [0xb4000077be1f0e10] [315] bytes parsed
Found transaction matching response.
Changing [client] [INVITE] transaction [0xb4000075b14fe1f0], from state [CALLING] to [PROCEEDING]
op [0xb4000076115dbc00] : setOrUpdateDialog() current=[0x0] new=[0x0]
Op [0xb4000076115dbc00] receiving call response [100], dialog is [0x0] in state [BELLE_SIP_DIALOG_NULL]
channel [0xb4000077be1f0e10]: ending recv background task with id=[46a2].
bellesip_wake_lock_release(): Android wake lock released [ref=0x46a2]
bellesip_wake_lock_acquire(): Android wake lock [belle-sip recv channel] acquired [ref=0x4a6e]
bellesip_wake_lock_acquire(): cast long of wakelock 19054
channel [0xb4000077be1f0e10]: starting recv background task with id=[4a6e].
channel [0xb4000077be1f0e10]: received [495] new bytes from [TCP://server:5060]:
SIP/2.0 180 Ringing
Via: SIP/2.0/TCP 192.168.2.19:41822;branch=z9hG4bK.7m2dpH0Mm;rport=41822;received=server
Contact: <sip:test2@server:5060;transport=TCP>
To: <sip:test2@server>;tag=0_590182924-765622192
From: <sip:test@server>;tag=Ls-v5FPWU
Call-ID: tSdngCajPw
CSeq: 21 INVITE
Date: Mon, 01 Apr 2024 13:40:49 GMT
User-Agent:  Mitel
P-Asserted-Identity: "T.Development"<sip:test2@server>
Content-Length: 0


channel [0xb4000077be1f0e10] [495] bytes parsed
Found transaction matching response.
New client dialog [0xb4000075c14d1010] , local tag [Ls-v5FPWU], remote tag [0_590182924-765622192]
Dialog [0xb4000075c14d1010]: now updated by transaction [0xb4000075b14fe1f0].
op [0xb4000076115dbc00] : setOrUpdateDialog() current=[0x0] new=[0xb4000075c14d1010]
Op [0xb4000076115dbc00] receiving call response [180], dialog is [0xb4000075c14d1010] in state [BELLE_SIP_DIALOG_EARLY]
[ToneManager] startRingbackTone
[ToneManager] destroyRingStream
This device (Google/Pixel 4a/sm6150) declares it has a built-in echo canceller.
No information available for [Google/Pixel 4a/sm6150],
Android >= 8 [API 33], using sound device descriptor.with DEVICE_HAS_BUILTIN_AEC | DEVICE_HAS_BUILTIN_OPENSLES_AEC flags
Sound device information for [Google/Pixel 4a/sm6150] is: builtin=[yes], delay=[0] ms
[AAudio Player] Created using device ID: AAudio Earpiece: Pixel 4a (2)
speex_lib_ctl init with neon ? 1
Configuring resampler input with rate=[8000], nchannels=[1]
MSFilePlayer[0xb400007581539c60]: /data/user/0/com.example.sipcall/files/share/sounds/linphone/ringback.wav opened: rate=8000,channel=1, length=1535 ms
configuring resampler output to rate=[48000], nchannels=[1]
ms_filter_link: MSFilePlayer:0xb400007581539c60,0-->MSDtmfGen:0xb400007581538550,0
ms_filter_link: MSDtmfGen:0xb400007581538550,0-->MSResample:0xb400007581539cf0,0
ms_filter_link: MSResample:0xb400007581539cf0,0-->MSAAudioPlayer:0xb40000758153bb50,0
Initializing speex resampler in mode [voip] from 1 channels
[AAudio Player] Using COMMUNICATION mode
[AAudio Player] Using device ID: AAudio Earpiece: Pixel 4a (2)
[AAudio Player] Player stream configured with samplerate 48000 and 1 channels
AAudioStreamBuilder_openStream() called ----------------------------------------
rate   =  48000, channels  = 1, channelMask = 0x80000001, format   = 1, sharing = EX, dir = OUTPUT
device =      2, sessionId = -1, perfMode = 12, callback: ON with frames = 0
usage  =      2, contentType = 1, inputPreset = 0, allowedCapturePolicy = 0
privacy sensitive = false, opPackageName = (null), attributionTag = (null)
Priority used: 99
Ring MSTicker priority increased to maximum.
PlayerBase::PlayerBase()
open() - openStream() returned -889, try switching from MONO to STEREO
PlayerBase::PlayerBase()
~AudioStreamInternal() 0xb4000076414e82c0 called
~AudioStream(s#10) mPlayerBase strongCount = 1
PlayerBase::~PlayerBase()
open(), request notificationFrames = -8, frameCount = 0
open() flags changed from 0x00000104 to 0x00000000
open() perfMode changed from 12 to 10
setState(s#11) from 0 to 2
AAudioStreamBuilder_openStream() returns 0 = AAUDIO_OK for s#11 ----------------
[AAudio Player] Player stream opened
[AAudio Player] Expected content type 1, got 1
[AAudio Player] Expected usage 2, got 2
AAudioStream_requestStart(s#11) called --------------
setState(s#11) from 2 to 3
AAudioStream_requestStart(s#11) returned 0 ---------
[AAudio Player] Player stream started
[AAudio Player] Current state is 3 / AAUDIO_STREAM_STATE_STARTING
onAudioDeviceUpdate(deviceId = 2)
[AAudio Player] Waited for state change, current state is 3 / AAUDIO_STREAM_STATE_STARTING (waited for 100 ms)
[AAudio Player] Asking for volume hack (lower & raise volume to workaround no sound on speaker issue, mostly on Samsung devices)
[Audio Manager] Lower & raise audio volume to workaround no sound issue until volume has changed...
[Audio Manager] Max volume for voice call stream is 7, current volume is 7
setState(s#11) from 3 to 4
CallSession [0xb400007581539eb8] moving from state LinphoneCallOutgoingProgress to LinphoneCallOutgoingRinging
Unable to find audio video conference with conference ID ConferenceId(peer=sip:test2@server, local=sip:test2@server) in RAM.
Linphone core [0xb4000076214e33d0] notified [call_state_changed]
channel [0xb4000077be1f0e10]: ending recv background task with id=[4a6e].
bellesip_wake_lock_release(): Android wake lock released [ref=0x4a6e]
Changing [client] [INVITE] transaction [0xb4000075b15b4810], from state [COMPLETED] to [TERMINATED]
Client INVITE transaction [0xb4000075b15b4810] terminated
transaction [0xb4000075b15b4810]: ending transaction background task with id=[2c02].
bellesip_wake_lock_release(): Android wake lock released [ref=0x2c02]
bellesip_wake_lock_acquire(): Android wake lock [belle-sip recv channel] acquired [ref=0x46a6]
bellesip_wake_lock_acquire(): cast long of wakelock 18086
channel [0xb4000077be1f0e10]: starting recv background task with id=[46a6].
channel [0xb4000077be1f0e10]: received [487] new bytes from [TCP://server:5060]:
OPTIONS sip:test@server:41822;transport=TCP SIP/2.0
Via: SIP/2.0/TCP server;branch=z9hG4bK-524287-1---4a5dc3646f4b7d6f;rport
Max-Forwards: 70
To: <sip:test@server:41822;transport=TCP>
From: <sip:server:5060>;tag=19108e13
Call-ID: ClWfurtCgBzUEhA9h7Cdtw..
CSeq: 6 OPTIONS
Allow: ACK,BYE,CANCEL,INVITE,NOTIFY,OPTIONS,REFER,REGISTER,SUBSCRIBE,INFO,PRACK,PUBLISH,UPDATE
User-Agent:  Mitel
Content-Length: 0


channel [0xb4000077be1f0e10] [487] bytes parsed
channel [0xb4000077be1f0e10]: message sent to [TCP://server:5060], size: [277] bytes
SIP/2.0 200 Ok
Via: SIP/2.0/TCP server;branch=z9hG4bK-524287-1---4a5dc3646f4b7d6f;rport
From: <sip:server:5060>;tag=19108e13
To: <sip:test@server:41822;transport=TCP>;tag=GpdsG
Call-ID: ClWfurtCgBzUEhA9h7Cdtw..
CSeq: 6 OPTIONS
Content-Length: 0
channel [0xb4000077be1f0e10]: ending recv background task with id=[46a6].
bellesip_wake_lock_release(): Android wake lock released [ref=0x46a6]
````

Note: Although push is enabled, all tests here were done with the app in the foreground shortly after registering/logging in the sip account, none of them were conducted in the background or without stable internet connection.

**Comment by Viish** [comment]
Please attach logs from both devices (A & B) from the start of the app to the moment the call is terminated on one device and the moment it times out on the other if it doesn't receive the BYE.
And please attach logs as files instead of text, it is way easier to follow the issue and read logs that way.

That being said, if you still have the issue in TCP, you should look in your proxy server logs as it probably is the culprit to why the BYE isn't propagated to the device.

**Comment by ParticleCore** [comment]
Conducting the same test with a different sip service results in the call terminating correctly as expected, I will assume the original issue is a problem tied to the sip server and not the sdk.
